File: frontend/src/types/competitor.ts
// 竞品分析相关类型 — 与后端 CompetitorVO / FeatureMatrixVO 字段保持一致

export type CompetitorType = 'direct' | 'indirect' | 'potential'
export type ThreatLevel = 'high' | 'medium' | 'low'
export type FeatureScore = 'yes' | 'partial' | 'no'

export interface Competitor {
  id: number
  projectId: number
  name: string
  logoUrl?: string
  website?: string
  type: CompetitorType
  region: string
  rating: number
  mau?: number
  pricingModel?: string
  fundingStage?: string
  threatLevel: ThreatLevel
  features: Record<string, FeatureScore>
  strengths: string[]
  weaknesses: string[]
  needsUserInput: boolean
  createdAt: string
  updatedAt: string
}

export interface FeatureMatrix {
  features: string[]
  scores: Record<string, FeatureScore[]>
  opportunities: string[]
  hasReport: boolean
}

export type InsightType = 'core_opportunity' | 'pricing_strategy' | 'warning'

export interface DifferentiationInsight {
  type: InsightType
  title: string
  description: string
}

export interface AnalyzeCompetitorsRequest {
  projectId: number
}

export interface AnalyzeCompetitorsResponse {
  taskId: number
  status: string
  estimatedSeconds: number
}

export type TaskStatus = 'pending' | 'running' | 'completed' | 'failed'

export interface CompetitorTaskInfo {
  taskId: number
  status: TaskStatus
  progress: number
  projectId: number
  errorMessage?: string
}

// 部分字段更新
export interface UpdateCompetitorPayload {
  name?: string
  logoUrl?: string
  website?: string
  type?: CompetitorType
  region?: string
  rating?: number
  mau?: number
  pricingModel?: string
  fundingStage?: string
  threatLevel?: ThreatLevel
  features?: Record<string, FeatureScore>
  strengths?: string[]
  weaknesses?: string[]
  needsUserInput?: boolean
}

File: backend/src/main/java/com/marketai/backend/entity/Competitor.java
package com.marketai.backend.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 单个竞品实体 — 一行一个竞品
 * JSON 列 (features/reviews/strengths/weaknesses) 以原始字符串存储, 由 Service 层解析
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@TableName("competitors")
public class Competitor {

    @TableId(type = IdType.AUTO)
    private Long id;

    private Long projectId;

    private String name;
    private String logoUrl;
    private String website;

    /** direct / indirect / potential */
    private String type;

    private String region;
    private BigDecimal rating;
    private Integer mau;
    private String pricingModel;
    private String fundingStage;

    /** high / medium / low */
    private String threatLevel;

    private String features;    // JSON: {"feature_name":"yes"|"partial"|"no"}
    private String reviews;     // JSON: 数组
    private String strengths;   // JSON: 数组
    private String weaknesses;  // JSON: 数组

    /** AI 标记的"需要用户补充"标志: 0=已知 / 1=不熟悉 */
    private Integer needsUserInput;

    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}
